### DISCUSSION TITLE
2 approaches for alleviating the IBC operational burden

### DISCUSSION THREAD

**Discussion by blewater** [e-money/em-ledger#101] [Ideas]
### Intro
The IBC protocol imposes significant operational requirements due to the exogenous proxy chain client setup:

1. Static handshake approach for establishing interchain communication.
2. Maintaining the clients in an updated state(receiving proof headers within a bonding period).
3. Running a Governance proposal as the remedy for unfreezing clients. Causes described here: [IBC Client Recovery](https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-026-ibc-client-recovery-mechanisms.md)*\** or when implementing a new IBC protocol.

#### Dynamic IBC
Agoric first identified this challenge as an IBC adoption hurdle ([the road to dynamic IBC](https://medium.com/agoric/the-road-to-dynamic-ibc-4a43bc964bca)) and advocated dynamic IBC a _refined_ IBC protocol that Confio adopted conceptually for the CosmWasm platform.
Thus **smart contracts** offer a more streamlined IBC operational approach that **does not** eliminate all operation strains but offer a helping hand toward smoother operations. We should evaluate CosmWasm for potential fit to our IBC use cases. Of interest is also their TS relayer and whether it offers an easier management path [ts-relayer features](https://github.com/confio/ts-relayer/blob/main/FEATURES.md).

#### Peer em-ledger IBC chain
Another approach emanating from the idea of [Cosmos as a single application chain](https://docs.cosmos.network/master/intro/why-app-specific.html) is creating a peer(s) chain for each IBC application with light client(s) attached to that single purpose chain. The benefits of this approach:
1. Seamless governance and chain upgrade whenever the IBC protocol/application(s) upgrade. 
2. Central operational rollout and administration of the IBC "components" by the e-Money organization. 
3. Improved security through isolation and supporting an external approval process to migrate IBC tokens to em-ledger. 

What do you think?

*\**
The issue with non-relaying funds and consequently locking them during the client freezing period has been slated for an SDK release: [pull #140](https://github.com/cosmos/ibc-go/pull/140)

**Comment by mdyring** [comment]
Interesting topic. :-)

Our top priority is deploying an IBC-enabled mainnet that can work with Cosmos Hub, Osmosis etc.

Given IBC is still "early" days, I think we should gain some operational experience with the existing protocol and tooling and then improve iteratively as needed.
